src/popover/use-one-time-popover.tsx

```diff
@@ -0,0 +1,39 @@
+import {useCallback, useEffect, useState} from 'react';
+import {storage, StorageModelKeysEnum} from '@atb/storage';
+import {PopOverKey} from './types';
+
+export const useOneTimePopover = () => {
+  const [seenPopovers, setSeenPopOvers] = useState<string[]>([]);
+
+  useEffect(() => {
+    getSeenPopOvers().then(setSeenPopOvers);
+  }, []);
+
+  /**
+   * Fetches from storage the list of one time popovers already seen by the user
+   */
+  const getSeenPopOvers = () =>
+    storage
+      .get(StorageModelKeysEnum.OneTimePopOver)
+      .then((stored) => JSON.parse(stored ?? '[]') as string[]);
+
+  const isSeen = useCallback(
```

Comment:
Here we se an example why I prefer specific naming. `isSeen` may be easy to understand here, but when used in other components, perhaps not. Also it sounds like a boolean instead of a function.

I would suggest returning `{getPopOverIsSeen, setPopOverIsSeen}` from this hook

Reply:
Chiming in here, with some pseudocode/pseudosituation 😅 

Generally I don't personally mind `const {isLoading} = useHelicopterState()`, since the helicopter part is implicit by the name of the used hook. I also think in a component like `HelicopterIcon` just using `const {isLoading} = useHelicopterState()` is enough, since that the `isLoading` variable is helicopter-related is understood from the surrounding context.

What we don't want is a large non-helicopter specific component that just uses the name `isLoading` throughout the code, and the context that it's the loading state for the helicopter is lost. In that situation a name like `isHelicopterLoading` is necessary. We can do that with `const {isLoading: isHelicopterLoading} = useHelicopterState()`, but then it is more up to the coder on a case by case basis to use a specific enough name. The option is of course (as suggested by Marius) to enforce it in the variable names so we always have `const {isHelicopterLoading} = useHelicopterState()`, which gives less room for not-explicit-enough named variables.

It might be nice to have a statement on this in the coding guideline, since we do this differently throughout the code. I don't really have a strong opinion on this personally, so we can just do what the team feels is better.

Reply:
Good thinking, would be nice to have some coding guidelines on this. Even though `const {isLoading: isHelicopterLoading} = useHelicopterState()` is possible, I think in most cases it is cleaner to enforce it in the return statement.
I do however see the benefits of a very short and simple variable name when its scope is limited internally to a function/component

Reply:
A lot of this comes down to personal preference. Personally I'm leaning towards explicitly naming properties when/if necessary, as in `const {isLoading: isHelicopterLoading} = useHelicopterState()`. In this case, where the hook is only used locally in the popover feature, and there is very little room for collisions with other hooks or variables with similar name, I prefer the short version.

One argument for renaming it is that the returns from this hook are inconsistently named. They should probably be either `{setPopOverSeen, isPopOverSeen}` or `{setSeen, isSeen}`. In that case, I prefer the first option.

Reply:
`{setPopOverSeen, isPopOverSeen}` sounds good :+1:

Reply:
As you say, it ultimately comes down to just personal preference, so having a coding guideline for it would help with consistency as @gorandalum suggested